src/eval/StateService.ml

```diff
@@ -153,14 +156,31 @@ module MakeStateService () = struct
           | Some _res' -> pure @@ res)
     | Local -> fetch_local ~fname ~keys fields
 
+  let fetch_bcinfo ~query_name ~query_args =
+    let%bind sm, _fields, _estates, bcinfo = assert_init () in
+    match sm with
+    | IPC socket_addr ->
+        StateIPCClient.fetch_bcinfo ~socket_addr ~query_name ~query_args
```

Comment:
As I understand it this does a bulk fetch of everything we deem to be "blockchain information". At the moment this is just the current block number, but we would need to fetch more information in the future.

Shouldn't we make the fetch more fine-grained, so that we can fetch "the current block number" or "the timestamp of block number X"?

Reply:
Oh, I see. I thought this fetched the entire contents of the hashtable. Never mind.